src/sparsezoo/objects/directory.py

```diff
@@ -284,9 +284,15 @@ def unzip(self, extract_directory: Optional[str] = None, force: bool = False):
             )
 
         name = ".".join(self.name.split(".")[:-2])
-        path = os.path.join(extract_directory, name)
-
-        if not os.path.exists(path) or force:  # do not re-unzip if not forced
+        is_model_gz = name == "model.onnx"
+        path = (
+            extract_directory if is_model_gz else os.path.join(extract_directory, name)
+        )
+        # if is_model_gz then path would point to the tarfile(s)
+        # parent directory and must be unzipped irrespective of existence
+        if (
+            is_model_gz or not os.path.exists(path) or force
```

Comment:
why do we always unzip if `is_model_gz`? what would happen if the tarball still exists but has already been unzipped?

Reply:
synced offline!